manifests/init.pp

```diff
@@ -217,33 +203,21 @@
     }
   }
 
-  if $puppet {
+  # We can't pull the certs out to the top level, because of how it gets the default
+  # parameter values from the main ::certs class.  Kafo can't handle that case, so
+  # it remains here for now.
+  if defined(Class['puppet']) {
     class { '::certs::puppet':
       hostname => $foreman_proxy_fqdn,
-    } ~>
-    class { '::puppet':
-      server                      => true,
-      server_ca                   => $::foreman_proxy::puppetca,
-      server_foreman_url          => $foreman_url,
-      server_foreman_ssl_cert     => $::certs::puppet::client_cert,
-      server_foreman_ssl_key      => $::certs::puppet::client_key,
-      server_foreman_ssl_ca       => $::certs::puppet::ssl_ca_cert,
-      server_storeconfigs_backend => false,
-      server_dynamic_environments => true,
-      server_environments_owner   => 'apache',
-      server_config_version       => '',
-      server_enc_api              => 'v2',
-      server_ca_proxy             => $puppet_ca_proxy,
-      server_implementation       => $puppet_server_implementation,
-      additional_settings         => {
-                                        'disable_warnings' => 'deprecations',
-      },
     }
+
+    Class['certs::puppet'] -> Class['puppet']
   }
 
   if $certs_tar {
     certs::tar_extract { $foreman_proxy_content::certs_tar: } -> Class['certs']
     Certs::Tar_extract[$certs_tar] -> Class['certs::foreman_proxy']
+    Certs::Tar_extract[$certs_tar] -> Class['certs::puppet']
```

Comment:
Won't this give a problem with strict definitions when puppet is not enabled? Maybe in the above puppet block you can chain ```Class['certs'] -> Class['certs::puppet']``` and since you already chain the extract before you implicitly get the same chaining.